### PR TITLE
rathole: 0.5.0 -> 0.5.0-unstable-2024-06-06

### PR DESCRIPTION
**Pull request by xokdvium**
## Description of changes

Fixes build failure introduced with Rust 1.80. See tracking issue: https://github.com/NixOS/nixpkgs/issues/332957.
The necessary time crate bump is already in main: https://github.com/rapiz1/rathole/issues/380

- Move to pkgs/by-name
- Update to [be14d12](https://github.com/rapiz1/rathole/commit/be14d12), which has Rust 1.80 fixes
- Re-enable doCheck
- Add myself as maintainer
- Sort meta attributes
- Remove with lib; usage
- Drop changelog for unstable version, since upstream does not track unreleased changes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by alyssais — DISMISSED** [review]
Please split this into multiple commits.

**Comment by xokdvium** [comment]
@ofborg eval

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/ra/rathole/package.nix [hunk 1]
```diff
@@ -30,18 +30,20 @@ rustPlatform.buildRustPackage rec {
 
   __darwinAllowLocalNetworking = true;
 
-  doCheck = false; # https://github.com/rapiz1/rathole/issues/222
+  doCheck = true;
```
Comment (alyssais) [inline pkgs/by-name/ra/rathole/package.nix:33]: This is the default.

**Review by alyssais — COMMENTED** [review]
In future when fixing a broken package, it'd be better to focus on just fixing the breakage, as it takes longer to find the time to review and merge a PR that makes a lot of changes.

**Comment by alienzj** [comment]
Hi, may I know can you all merge it?

**Pull request merged**